flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/SystemTypeInference.java

```diff
@@ -135,6 +142,25 @@ private static void checkReservedArgs(List<StaticArgument> staticArgs) {
         }
     }
 
+    private static void checkMultipleTableArgs(List<StaticArgument> staticArgs) {
+        if (staticArgs.stream().filter(arg -> arg.is(StaticArgumentTrait.TABLE)).count() > 1) {
```

Comment:
```suggestion
        if (staticArgs.stream().filter(arg -> arg.is(StaticArgumentTrait.TABLE)).limit(2).count() > 1) {
```
nit: would it make sense to add limit like?

Reply:
given that in most cases this is just 2 or 3, limiting make the code just more complex